controller/node_controller.go

```diff
@@ -435,6 +399,52 @@ func (nc *NodeController) syncNode(key string) (err error) {
 
 	}
 
+	// sync node state by manager pod
+	managerPods, err := nc.ds.ListManagerPods()
+	if err != nil {
+		return err
+	}
+	nodeManagerFound := false
+	for _, pod := range managerPods {
+		if pod.Spec.NodeName != node.Name {
+			continue
+		}
+		nodeManagerFound = true
+		if !kubeNodeIsReady {
+			break
+		}
```

Comment:
Don't think you need this, but rather let the below logic to update the node condition.